thriftpy/parser/parser.py

```diff
@@ -336,3 +447,229 @@ def _parse_seq(p):
         p[0] = [p[1]] + p[2]
     elif len(p) == 1:
         p[0] = []
+
+
+def _cast(t):  # noqa
+    if t == TType.BOOL:
+        return _cast_bool
+    if t == TType.BYTE:
+        return _cast_byte
+    if t == TType.I16:
+        return _cast_i16
+    if t == TType.I32:
+        return _cast_i32
+    if t == TType.I64:
+        return _cast_i64
+    if t == TType.DOUBLE:
+        return _cast_double
+    if t == TType.STRING:
+        return _cast_string
+    if t == TType.BINARY:
+        return _cast_binary
+    if t[0] == TType.LIST:
+        return _cast_list(t)
+    if t[0] == TType.SET:
+        return _cast_set(t)
+    if t[0] == TType.MAP:
+        return _cast_map(t)
+    if t[0] == TType.I32:
+        return _cast_enum(t)
+    if t[0] == TType.STRUCT:
+        return _cast_struct(t)
+
+
+def _cast_bool(v):
+    assert isinstance(v, bool)
+    return v
+
+
+def _cast_byte(v):
+    assert isinstance(v, str)
+    return v
+
+
+def _cast_i16(v):
+    assert isinstance(v, int)
+    return v
+
+
+def _cast_i32(v):
+    assert isinstance(v, int)
+    return v
+
+
+def _cast_i64(v):
+    assert isinstance(v, int)
+    return v
+
+
+def _cast_double(v):
+    assert isinstance(v, float)
+    return v
+
+
+def _cast_string(v):
+    assert isinstance(v, str)
+    return v
+
+
+def _cast_binary(v):
+    assert isinstance(v, str)
+    return v
+
+
+def _cast_list(t):
+    assert t[0] == TType.LIST
+
+    def __cast_list(v):
+        assert isinstance(v, list)
```

Comment:
Yep, this is just a matter of preference.

Apache thrift enables this: `list <i32> lst = '123456'`.